test/parallel/test-fs-write-sync-optional-params.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+require('../common');
+const fs = require('fs');
+const path = require('path');
+const tmpdir = require('../common/tmpdir');
+const assert = require('assert');
+
+tmpdir.refresh();
+
+const dest = path.resolve(tmpdir.path, 'tmp.txt');
+const buffer = Buffer.from('zyx');
+
+{
+  let fd = fs.openSync(dest, 'w+');
```

Comment:
It would be useful to wrap all the testing in `try`/`finally` block so we don't leave open fd if the tests fails. Creating a `test` function could help. Same thing for the async tests.

```suggestion
  let fd
  try {
    fd = fs.openSync(dest, 'w+')
    test(fd);
  } finally {
    if (fd != null) fs.closeSync(fd)
  }
```

Reply:
Thanks for suggestion!
I've wrapped the whole "open, test, close" sequence in functions to ensure fresh fd/fh for each test. This will prevent any uncertainty when `typeof position !== 'number'`, so current position won't be affected by previous iterations.